core/src/main/scala/org/apache/spark/executor/TaskMetrics.scala

```diff
@@ -39,68 +37,144 @@ import org.apache.spark.util.Utils
 
   /**
-   * The number of in-memory bytes spilled by this task
+   * The number of in-memory bytes spilled by this task.
    */
-  private var _memoryBytesSpilled: Long = _
-  def memoryBytesSpilled: Long = _memoryBytesSpilled
-  private[spark] def incMemoryBytesSpilled(value: Long): Unit = _memoryBytesSpilled += value
-  private[spark] def decMemoryBytesSpilled(value: Long): Unit = _memoryBytesSpilled -= value
+  def memoryBytesSpilled: Long = _memoryBytesSpilled.localValue
 
   /**
-   * The number of on-disk bytes spilled by this task
+   * The number of on-disk bytes spilled by this task.
    */
-  private var _diskBytesSpilled: Long = _
-  def diskBytesSpilled: Long = _diskBytesSpilled
-  private[spark] def incDiskBytesSpilled(value: Long): Unit = _diskBytesSpilled += value
-  private[spark] def decDiskBytesSpilled(value: Long): Unit = _diskBytesSpilled -= value
+  def diskBytesSpilled: Long = _diskBytesSpilled.localValue
+
+  /**
+   * Peak memory used by internal data structures created during shuffles, aggregations and
+   * joins. The value of this accumulator should be approximately the sum of the peak sizes
+   * across all such data structures created in this task. For SQL jobs, this only tracks all
+   * unsafe operators and ExternalSort.
+   */
+  def peakExecutionMemory: Long = _peakExecutionMemory.localValue
+
+  /**
+   * Storage statuses of any blocks that have been updated as a result of this task.
+   */
+  def updatedBlockStatuses: Seq[(BlockId, BlockStatus)] = _updatedBlockStatuses.localValue
+
+  @deprecated("use updatedBlockStatuses instead", "2.0.0")
+  def updatedBlocks: Option[Seq[(BlockId, BlockStatus)]] = {
+    if (updatedBlockStatuses.nonEmpty) Some(updatedBlockStatuses) else None
+  }
+
+  // Setters and increment-ers
+  private[spark] def setExecutorDeserializeTime(v: Long): Unit =
+    _executorDeserializeTime.setValue(v)
+  private[spark] def setExecutorRunTime(v: Long): Unit = _executorRunTime.setValue(v)
+  private[spark] def setResultSize(v: Long): Unit = _resultSize.setValue(v)
+  private[spark] def setJvmGCTime(v: Long): Unit = _jvmGCTime.setValue(v)
+  private[spark] def setResultSerializationTime(v: Long): Unit =
+    _resultSerializationTime.setValue(v)
+  private[spark] def incMemoryBytesSpilled(v: Long): Unit = _memoryBytesSpilled.add(v)
+  private[spark] def incDiskBytesSpilled(v: Long): Unit = _diskBytesSpilled.add(v)
+  private[spark] def incPeakExecutionMemory(v: Long): Unit = _peakExecutionMemory.add(v)
+  private[spark] def incUpdatedBlockStatuses(v: Seq[(BlockId, BlockStatus)]): Unit =
+    _updatedBlockStatuses.add(v)
+  private[spark] def setUpdatedBlockStatuses(v: Seq[(BlockId, BlockStatus)]): Unit =
+    _updatedBlockStatuses.setValue(v)
+
+  /**
+   * Host's name the task runs on.
+   */
+  @deprecated("hostname will be removed from TaskMetrics in the future", "2.0.0")
+  def hostname: String = Utils.localHostName()
```

Comment:
Before, the hostname was the hostname of the machine which produced the metrics, whereas it looks like this is changing the behavior to return the hostname of the consumer. If anyone had code which actually relied on these hostnames, won't this change risk silently breaking their code? Just wondering if it would be better to remove this altogether if the alternative is a change in semantics.